zkevm-circuits/src/evm_circuit/execution/precompiles/sha256.rs

```diff
@@ -0,0 +1,259 @@
+            region
+                .challenges()
+                .keccak_input()
+                .map(|r| rlc::value(input_bytes.iter().rev(), r));
+
+            region
+                .challenges()
+                .keccak_input()
+                .map(|r| rlc::value(output_bytes.iter().rev(), r));
+
+            self.input_bytes_rlc.assign(
+                region,
+                offset,
+                region
+                    .challenges()
+                    .keccak_input()
+                    .map(|r| rlc::value(input_bytes.iter().rev(), r)),
+            )?;
+            self.output_bytes_rlc.assign(
+                region,
+                offset,
+                region
+                    .challenges()
+                    .keccak_input()
+                    .map(|r| rlc::value(output_bytes.iter().rev(), r)),
+            )?;
+            self.return_bytes_rlc.assign(
+                region,
+                offset,
+                region
+                    .challenges()
+                    .keccak_input()
+                    .map(|r| rlc::value(return_bytes.iter().rev(), r)),
+            )?;
+        } else {
+            log::error!("unexpected aux_data {:?} for sha256", step.aux_data);
+            return Err(Error::Synthesis);
+        }
+        self.input_word_size.assign(
+            region,
+            offset,
+            (call.call_data_length + (N_BYTES_WORD as u64) - 1).into(),
+        )?;
+        self.is_success.assign(
+            region,
+            offset,
+            Value::known(F::from(u64::from(call.is_success))),
+        )?;
+        self.callee_address.assign(
+            region,
+            offset,
+            Value::known(call.code_address.unwrap().to_scalar().unwrap()),
+        )?;
+        self.caller_id
+            .assign(region, offset, Value::known(F::from(call.caller_id as u64)))?;
+        self.call_data_offset.assign(
+            region,
+            offset,
+            Value::known(F::from(call.call_data_offset)),
+        )?;
+        self.call_data_length.assign(
+            region,
+            offset,
+            Value::known(F::from(call.call_data_length)),
+        )?;
+        self.return_data_offset.assign(
+            region,
+            offset,
+            Value::known(F::from(call.return_data_offset)),
+        )?;
+        self.return_data_length.assign(
+            region,
+            offset,
+            Value::known(F::from(call.return_data_length)),
+        )?;
+        self.restore_context
+            .assign(region, offset, block, call, step, 7)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use bus_mapping::{
+        evm::{OpcodeId, PrecompileCallArgs},
+        precompile::PrecompileCalls,
+    };
+    use eth_types::{bytecode, ToWord};
+    use itertools::Itertools;
+    use mock::TestContext;
+
+    use crate::test_util::CircuitTestBuilder;
+
+    lazy_static::lazy_static! {
+        static ref TEST_VECTOR: Vec<PrecompileCallArgs> = {
```

Comment:
more case is added. However, calldata length for some edge cases (like multiple of 32) may not work as it expected since the SHA256 circuit is not involved in precompile's unittest (sha256 is filled by `dev_load`).

There are tests in repo of SHA256 circuit for hashed length with edge cases